### PR TITLE
workflows: Log more detailed errors so we can debug why workflows sometimes don't run

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/server/workflow/service/service.go [hunk 1]
```diff
@@ -542,6 +541,7 @@ func (ws *workflowService) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	webhookID := workflowMatch[1]
 	if err := ws.startWorkflow(webhookID, r); err != nil {
+		log.Warningf("Failed to start workflow: %s", err)
```
Comment (tylerwilliams) [inline enterprise/server/workflow/service/service.go:544]: log.Error might be more appropriate here?

File: enterprise/server/workflow/service/service.go [hunk 2]
```diff
@@ -489,29 +489,28 @@ func (ws *workflowService) startWorkflow(webhookID string, r *http.Request) erro
...
-		return err
+		return status.WrapError(err, "parse webhook request")
 	}
 	if webhookData == nil {
 		return nil
 	}
 	wf, err := ws.readWorkflowForWebhook(ctx, webhookID)
 	if err != nil {
-		return err
+		return status.WrapError(err, "lookup workflow")
```
Comment (tylerwilliams) [inline enterprise/server/workflow/service/service.go:499]: I don't personally love using wrapError everywhere. I think it's pretty noisy for little gain. Do we need these?

**Review by tylerwilliams — APPROVED** [review]
LGTM

File: enterprise/server/workflow/service/service.go (hunk 2 above)
Comment (bduffany) [reply enterprise/server/workflow/service/service.go:499]: Discussed offline a bit, not sure yet whether these will add much value beyond saving a little bit of mental energy figuring out where an error came from. Once we have a way to expose a stack trace from an error, that will be a more scalable solution.

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**